# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1245389359 -7200
#      Fri Jun 19 07:29:19 2009 +0200
# Node ID da465c405d849d37651a9b9d08e1ef94cf6720e9
# Parent  2ebf3ca62add9ae8c8a60df71fcaaccf41510850
fix small defect in oct_sort::lookup

diff --git a/liboctave/oct-sort.cc b/liboctave/oct-sort.cc
--- a/liboctave/oct-sort.cc
+++ b/liboctave/oct-sort.cc
@@ -1898,17 +1898,17 @@ octave_sort<T>::lookup (const T *data, o
       int nbadruns = 0;
       while (vlo != nvalues && nbadruns < 16)
         {
           octave_idx_type vhi;
 
           // Determine a sorted run.
           for (vhi = vlo + 1; vhi != nvalues; vhi++)
             {
-              if (! comp (values[vhi-1], values[vhi]))
+              if (comp (values[vhi], values[vhi-1]))
                 break;
             }
 
           // Do a recursive lookup.
           lookup_merge (data - offset, offset, nel + offset,
                         values + vlo, vhi - vlo, idx + vlo, comp);
 
           if (vhi - vlo <= 2)
